test/e2e/tests/test_observability.py

```
@@ -0,0 +1,1482 @@
+                mismatches.append(
+                    f"  {metric_name}: expected '{expected_type}', got '{actual_type}'"
+                )
+            else:
+                print(f"[type] {metric_name}: {actual_type} ✓")
+
+        if skipped:
+            print(f"[type] Skipped (no metadata): {', '.join(skipped)}")
+
+        if mismatches:
+            pytest.fail(
+                f"FAIL: Metric type mismatches found:\n"
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)